custom_components/nrgkick/number.py

```diff
@@ -50,6 +53,18 @@ class NRGKickNumberEntityDescription(NumberEntityDescription, NRGKickRequiredKey
         value_fn=lambda data: data["cc_s"].chargeCurrent.userSet,
         api_fn=lambda c, v: c.set_charge_current_limit(v),
     ),
+    NRGKickNumberEntityDescription(
+        key="energy_limit_value",
+        translation_key="energy_limit_value",
+        device_class=NumberDeviceClass.CURRENT,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        native_max_value= lambda data: data["cc_s"].energyLimit.max,
+        native_min_value= lambda data: data["cc_s"].energyLimit.min,
+        native_step=0.5,
```

Comment:
In the App itself I can only change in steps of 1kWh, you tested that 0.5 is possible?

Reply:
No i did not tested it yet.
I will set the native_step to 1.0